### PR TITLE
media: Add Encoder::create() 

### PR DESCRIPTION
**Pull request by junmin-kim**

### CODE REVIEW

**Comment by junmin-kim** [comment]
@jiawei-xiang I think Decoder and Encoder should have same interface. Would you review this PR?

File: framework/src/media/Encoder.cpp [hunk 1]
```diff
@@ -26,18 +26,53 @@
...
+	auto instance = std::make_shared<Encoder>(audioType, channels, sampleRate);
+	if (instance && instance->init()) {
+		return instance;
+	} else {
+		meddbg("%s[line : %d] Fail : init is failed\n", __func__, __LINE__);
+		meddbg("audioType : %d, channels : %d, sampleRate : %d\n", audioType, channels, sampleRate);
+		return nullptr;
+	}
+
+	meddbg("Error! (%d,%d,%d), failed!\n", audioType, channels, sampleRate);
```
Comment (JeongJunSik) [inline framework/src/media/Encoder.cpp]: It does not call here. because of above if or else.
It is a dead code.

Comment (junmin-kim) [reply framework/src/media/Encoder.cpp]: Fixed it

**Comment by seinfra** [comment]
Target : [63aeb93f6ac5354b2bafc0f04226aeac49322fe8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [63aeb93f6ac5354b2bafc0f04226aeac49322fe8] - Code Rule Check Result:
framework/src/media/Encoder.h:40: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/Encoder.h:40: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/Encoder.h:40: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/Encoder.h:40: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)

**Comment by seinfra** [comment]
Target : [a6590a4a565993fb89a8cf70b92f086f89f20d17] - Code Rule Check Result:
framework/src/media/Encoder.h:40: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/Encoder.h:40: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/Encoder.h:40: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/Encoder.h:40: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)

**Comment by seinfra** [comment]
Target : [a6590a4a565993fb89a8cf70b92f086f89f20d17] - Code Rule Check (C++) OK.

**Pull request merged**